services/webhook/deliver.go

```diff
@@ -175,7 +172,7 @@ func Deliver(t *webhook_model.HookTask) error {
 		return fmt.Errorf("Webhook task skipped (webhooks disabled): [%d]", t.ID)
 	}
 
-	resp, err := webhookHTTPClient.Do(req.WithContext(context.WithValue(req.Context(), contextKeyWebhookRequest, req)))
```

Comment:
DoWithContext()

Reply:
(let a request be cancable by hammer)

Reply:
The original code in 0e7f7df3cf176640c66ddf286ec052c7c13beb8a was `resp, err := webhookHTTPClient.Do(req)`, and then I changed it to the complex line, then now the complex line is reverted to the original code.

I think it won't be a problem.

If you want to use `DoWithContext`, what context should be used?

Reply:
the one obtained from the gracefull shutdown ... @zeripath did contribute :)

Reply:
There are many HTTP related codes don't use the graceful shutdown context. I doubt about if it is really necessary.

If it is necessary, we should change all legacy code to the Do-with-Context, and write it into guideline.

For example:

![image](https://user-images.githubusercontent.com/2114189/142553511-2533af75-9915-46df-8066-8cf4cb993e3b.png)

![image](https://user-images.githubusercontent.com/2114189/142553447-f8a09f57-a5bf-4dde-9faa-6aa26275f823.png)

![image](https://user-images.githubusercontent.com/2114189/142553126-1d7e73d7-7734-44a7-97eb-30a00f89d94e.png)

Reply:
I just modified the webhook code to `webhookHTTPClient.Do(req.WithContext(graceful.GetManager().ShutdownContext()))`

Please advise.